neaten/neaten_test.go

```diff
@@ -0,0 +1,333 @@
+	srcUniquePerl := "cci4fafnu1ia052l75tg"
+
+	Convey("Given existing source and dest dirs you can tidy the source", t, func() {
+		tmpDir := t.TempDir()
+		srcDir := filepath.Join(tmpDir, "src", srcUniversal)
+		destDir := filepath.Join(tmpDir, "dest", "final")
+		interestUniqueDir1 := createTestPath([]string{srcDir, "go", srcUniqueGo})
+		interestUniqueDir2 := createTestPath([]string{srcDir, "perl", srcUniquePerl})
+
+		combineSuffixes, dbSuffixes, baseSuffixes := buildSrcDir(srcDir, srcUniqueGo, srcUniquePerl,
+			interestUniqueDir1, interestUniqueDir2)
+
+		createTestDirWithDifferentPerms(destDir)
+
+		test := Tidy{
+			SrcDir:  srcDir,
+			DestDir: destDir,
+			Date:    date,
+
+			CombineFileSuffixes: combineSuffixes,
+			DBFileSuffixes:      dbSuffixes,
+			BaseFileSuffixes:    baseSuffixes,
+
+			CombineFileGlobPattern:  "%s/*/*/%s",
+			DBFileGlobPattern:       "%s/*/*/%s",
+			WalkFilePathGlobPattern: "%s/*/*/*%s",
+
+			DestDirPerms: modePermUser,
+		}
+
+		err := test.Up()
+
+		Convey("And the combine files are moved from the source dir to the dest dir", func() {
+			combineFileSuffixes := [4]string{".logs.gz", ".byusergroup.gz", ".bygroup", ".stats.gz"}
+
+			for i := range combineFileSuffixes {
+				final1 := filepath.Join(destDir, date+"_go."+srcUniqueGo+"."+srcUniversal+combineFileSuffixes[i])
+				_, err = os.Stat(final1)
+				So(err, ShouldBeNil)
+
+				final2 := filepath.Join(destDir, date+"_perl."+srcUniquePerl+"."+srcUniversal+combineFileSuffixes[i])
+				_, err = os.Stat(final2)
+				So(err, ShouldBeNil)
+			}
+		})
+
+		Convey("And the the contents of the .basedirs and .dgut.dbs dir exist", func() {
+			dbsPath := filepath.Join(destDir, date+"_"+srcUniversal)
+			dbsSuffixes := [5]string{
+				".basedirs",
+				".dgut.dbs/0/dgut.db",
+				".dgut.dbs/0/dgut.db.children",
+				".dgut.dbs/1/dgut.db",
+				".dgut.dbs/1/dgut.db.children"}
+
+			for i := range dbsSuffixes {
+				_, err = os.Stat(dbsPath + dbsSuffixes[i])
+				So(err, ShouldBeNil)
+			}
+		})
+
+		Convey("And the .dgut.dbs.updated file exists in the dest dir", func() {
+			expectedFileName := filepath.Join(destDir, ".dgut.dbs.updated")
+
+			_, err = os.Stat(expectedFileName)
+			So(err, ShouldBeNil)
+		})
+
+		Convey("And the mtime of the .dgut.dbs file matches the oldest mtime of the walk log files", func() {
+			err = os.RemoveAll(tmpDir)
+			So(err, ShouldBeNil)
+
+			buildSrcDir(srcDir, srcUniqueGo, srcUniquePerl, interestUniqueDir1, interestUniqueDir2)
+
+			createTestDirWithDifferentPerms(destDir)
+
+			newMtimeFile := filepath.Join(interestUniqueDir1, "walk.1.log")
+			mTime := time.Date(2006, time.April, 1, 3, 4, 5, 0, time.UTC)
+			aTime := time.Date(2007, time.March, 2, 4, 5, 6, 0, time.UTC)
+
+			err = os.Chtimes(newMtimeFile, aTime, mTime)
+			So(err, ShouldBeNil)
+
+			err = test.Up()
+			So(err, ShouldBeNil)
+
+			dbsFileMTime := getMTime(filepath.Join(destDir, ".dgut.dbs.updated"))
+
+			So(mTime, ShouldEqual, dbsFileMTime)
```

Comment:
This is backwards; the test result should come first, and the expected result should come after the Should*.

Reply:
I think it's the correct order no? Source dir is built, -> dest dir is built, -> very old a and m times are assigned to a file called newMtimeFile, -> up is ran, -> mTime of the .dgut.dbs.updated is found (the file whose mTime should be equal to the oldest of all the files), -> tests that is equal to the arbitrary mTime assigned at the beginning

Reply:
The format is `So(valueToTest, ShouldEqual, expectedValue)`. dbsFileMTime is the value you're trying to test, since it's the mTime of a file that the function under test, `Up`, touched. mTime is the expected value, since it's the one we manually set for testing purposes.

So it should be `So(dbsFileMtime, ShouldEqual, mTime)`.

Reply:
(You can make it clearer by giving these better variable names, like mTime -> expectedMTime)

Reply:
Okay I see